hles/transformation/src/main/scala/org/broadinstitute/monster/dap/environment/TemperaturePrecipitationTransformations.scala

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.monster.dap.environment
+
+import org.broadinstitute.monster.dap.RawRecord
+import org.broadinstitute.monster.dogaging.jadeschema.fragment.EnvironmentTemperaturePrecipitation
+
+object TemperaturePrecipitationTransformations {
+
+  /**
+    * Parse all temperature and precipitation variables out of a raw RedCap record,
+    * injecting them into a partially-modeled environment record.
+    */
+  def mapTemperaturePrecipitationVariables(
+    rawRecord: RawRecord
+  ): EnvironmentTemperaturePrecipitation = {
+    EnvironmentTemperaturePrecipitation(
+      tpDataYear = rawRecord.getOptionalNumber("tp_data_year"),
+      tpPcpnAnnual01 = rawRecord.getOptional("tp_pcpn_annual_01").map(_.toDouble),
+      tpPcpnAnnual02 = rawRecord.getOptional("tp_pcpn_annual_02").map(_.toDouble),
+      tpPcpnAnnual03 = rawRecord.getOptional("tp_pcpn_annual_03").map(_.toDouble),
+      tpPcpnAnnual04 = rawRecord.getOptional("tp_pcpn_annual_04").map(_.toDouble),
+      tpPcpnAnnual05 = rawRecord.getOptional("tp_pcpn_annual_05").map(_.toDouble),
+      tpPcpnAnnual06 = rawRecord.getOptional("tp_pcpn_annual_06").map(_.toDouble),
+      tpPcpnAnnual07 = rawRecord.getOptional("tp_pcpn_annual_07").map(_.toDouble),
+      tpPcpnAnnual08 = rawRecord.getOptional("tp_pcpn_annual_08").map(_.toDouble),
+      tpPcpnAnnual09 = rawRecord.getOptional("tp_pcpn_annual_09").map(_.toDouble),
+      tpPcpnAnnual10 = rawRecord.getOptional("tp_pcpn_annual_10").map(_.toDouble),
+      tpPcpnAnnual11 = rawRecord.getOptional("tp_pcpn_annual_11").map(_.toDouble),
+      tpPcpnAnnual12 = rawRecord.getOptional("tp_pcpn_annual_12").map(_.toDouble),
```

Comment:
is it important that we define these individually in sequence, or is there some way we could just pass in all of these values in a loop for each month, 1-12?

Reply:
This could get sticky; we codegen (via a plugin from `ingest-utils`) the underlying case class `EnvironmentTemperaturePrecipitation` from a JSON schema. We could muck with the case class JSON encoder to do the array transformation as you allude to, but it'd get clobbered on the next compile.